app/src/main/java/com/ncnf/map/MapHandler.java

```diff
@@ -0,0 +1,244 @@
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+
+            }
+        });
+    }
+
+    //Creates the behavior for when clicking on a suggestion
+    private void createSuggestionsClickListener(MaterialSearchBar materialSearchBar, PlacesClient placesClient){
+        materialSearchBar.setSuggestionsClickListener(new SuggestionsAdapter.OnItemViewClickListener() {
+            @Override
+            public void OnItemClickListener(int position, View v) {
+                if (position >= predictionList.size())
+                    return;
+                AutocompletePrediction selection = predictionList.get(position);
+                //Stores the address as a human-readable String in 'suggestion'
+                String suggestion = materialSearchBar.getLastSuggestions().get(position).toString();
+                materialSearchBar.setText(suggestion);
+
+                //Because apparently just putting mSB.clearSuggestions() does not work
+                new Handler().postDelayed(materialSearchBar::clearSuggestions, 1000);
+
+                //Hides the keyboard
+                InputMethodManager input = (InputMethodManager) context.getSystemService(INPUT_METHOD_SERVICE);
+                if (input != null)
+                    input.hideSoftInputFromWindow(materialSearchBar.getWindowToken(), InputMethodManager.HIDE_IMPLICIT_ONLY);
+
+                //Prepares the conversion from prediction to LatLng
+                final String placeId = selection.getPlaceId();
+                List<Place.Field> placeFields = Arrays.asList(Place.Field.LAT_LNG);
+
+                //Converts prediction to LatLng and updates markers if successful, prints out errors if not
+                FetchPlaceRequest fetchPlaceRequest = FetchPlaceRequest.builder(placeId, placeFields).build();
+                placesClient.fetchPlace(fetchPlaceRequest).addOnSuccessListener(fetchPlaceResponse -> {
+                    Place place = fetchPlaceResponse.getPlace();
+                    Log.i("Place fetching", "Place found: " + place.getName());
+                    LatLng placeLatLng = place.getLatLng();
+                    if (placeLatLng != null) {
+                        userPosition = placeLatLng;
+                        update_markers();
+                    }
+                }).addOnFailureListener(e -> {
+                    if (e instanceof ApiException){
+                        ApiException apiException = (ApiException) e;
+                        apiException.printStackTrace();
+                        int statusCode = apiException.getStatusCode();
+                        Log.i("Place fetching", "Place not found: " + e.getMessage());
+                        Log.i("Place fetching", "Status code: " + statusCode);
+                    }
+                });
+            }
+
+            @Override
+            public void OnItemDeleteListener(int position, View v) {
+
+            }
+        });
+    }
```

Comment:
@Arthelh What do you think about creating a SearchBarHandler which contains these 3 functions in order to separate the Map from the SearchBar elements and modularize ? This Search bar may be useful some place else, in the settings for example to set the position I'm at once for all, etc... Tell me if you think it's a good idea !

Reply:
That's definitely something to think about. We can do it in a later branch. I was also thinking once the feedView would be completely implemented, we could use the searchBar to filter events by tags or other too!

Reply:
Yep that's my point, I'll modularize that in my branch which will use fragments for most of the activities.